### PR TITLE
feat(ui): add network table checkboxes

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add checkboxes to the network table in machine details.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit the network tab in machine details for a new or ready machine.
- Check that there are checkboxes and that they work as expected.
- Visit a deployed machine and check that the checkboxes are disabled.

## Fixes

Fixes: canonical-web-and-design/maas-squad#2284.

### CODE REVIEW

**Review by Caleb-Ellis — COMMENTED** [review]
The checkboxes are correctly not showing up for the parent interfaces, but they're still being factored into the group checkbox behaviour, i.e. manually selecting all child interfaces will have the group checkbox showing a `-` even though everything that *can* be selected, is selected. Similarly, checking the group checkbox makes it a tick, which probably means the parent interface id is getting included in selected state.

As an aside it looks like the table headers have messed up at some point. This happened to me too in the storage tab when making one of the tables expanding, which turns the headers into flex elements. Wrapping the double row into a single container div should fix it.

**Comment by huwshimi** [comment]
> The checkboxes are correctly not showing up for the parent interfaces, but they're still being factored into the group checkbox behaviour, i.e. manually selecting all child interfaces will have the group checkbox showing a `-` even though everything that _can_ be selected, is selected. Similarly, checking the group checkbox makes it a tick, which probably means the parent interface id is getting included in selected state.

Thanks, that should be fixed now.

> As an aside it looks like the table headers have messed up at some point. This happened to me too in the storage tab when making one of the tables expanding, which turns the headers into flex elements. Wrapping the double row into a single container div should fix it.

I have a fix for this in another branch (next PR). It was making it an expanding table that broke it, just like your situation.

File: ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTable.tsx [hunk 1]
```diff
@@ -500,6 +542,11 @@ const NetworkTable = ({ systemId }: Props): JSX.Element => {
         {
           content: (
             <>
+              <GroupCheckbox
+                items={nicIds}
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTable.tsx:546]: Maybe this should include the disabled prop if networking is disabled?

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Just one comment for consideration

File: ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTable.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTable.tsx:546]: Oops yes it needs that!

**Pull request merged**